components/p3a/message_manager.cc

```diff
@@ -122,6 +122,15 @@ void MessageManager::Init(
   }
 }
 
+void MessageManager::Stop() {
```

Comment:
we also check the preference in `StartScheduledUpload` and friends, maybe we can remove that check now or make it `CHECK` ?

Reply:
`CHECK` sounds good